server.js

```diff
@@ -1,32 +1,103 @@
-import express from 'express'
+import express, { response } from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+import avocadoSalesData from './data/avocado-sales.json'
+
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+console.log("this is node working")
+
+
+// Start defining routes
+app.get('/', (request, response) => {
+  response.send('Hello, welcome to my api! Find some information about avocado sales in the US here. ')
+})
+
+//plural data returned endpoint
+app.get('/sales', (request, response) => {
+  const { date } = request.query;
+  const { region } = request.query;
+  if (date) {
+    const salesByDate = avocadoSalesData.filter((item) => item.date === date);
+    if(salesByDate.length > 0) {
+      return response.json(salesByDate)
+    } else if (salesByDate.length === 0) {
+      return response.status(404).json("ERROR_NO_SALES_BY_THIS_DATE_FOUND")
```

Comment:
Error messages that we send to the front-end should be more descriptive - `"ERROR_NO_SALES_BY_THIS_DATE_FOUND"` is a great name for variable storing message, but not for the message itself. Maybe consider something like :

```
const ERROR_NO_SALES_BY_THIS_DATE_FOUND = {
    message: "No records found by date"
}
```